crates/polars-io/src/path_utils/hugging_face.rs

```diff
@@ -31,17 +38,17 @@ struct HFRepoLocation {
 
 impl HFRepoLocation {
     fn new(bucket: &str, repository: &str, revision: &str) -> Self {
-        let bucket = percent_encode(bucket.as_bytes());
-        let repository = percent_encode(repository.as_bytes());
-
-        // "https://huggingface.co/api/ [datasets | spaces] / {username} / {reponame} / tree / {revision} / {path from root}"
+        // Don't percent-encode bucket/repository/revision - they are path segments
```

Comment:
I think you should percent-encode revisions, e.g. `refs/convert/parquet` -> `refs%2F/convert%2Fparquet` in the final URL